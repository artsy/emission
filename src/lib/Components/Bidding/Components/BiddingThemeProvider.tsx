import React from "react"
import { ThemeProvider } from "styled-components"
import { theme } from "../Elements/Theme"

export class BiddingThemeProvider extends React.Component {
  render() {
    return <ThemeProvider theme={theme} {...this.props} />
  }
}
