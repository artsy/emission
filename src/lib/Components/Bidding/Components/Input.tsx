import React, { Component } from "react"
import { TextInputProperties } from "react-native"
import { TextInput, TextInputProps } from "../Elements/TextInput"

interface InputProps extends TextInputProps, TextInputProperties {
  error?: boolean
  inputRef?: (component: any) => void
}

class InputState {
  borderColor: "black10" | "purple100" | "red100"
}

export class Input extends Component<InputProps, InputState> {
  constructor(props) {
    super(props)

    this.state = {
      borderColor: this.props.error ? "red100" : "black10",
    }
  }

  componentDidUpdate(prevProps) {
    if (this.props.error !== prevProps.error) {
      this.setState({ borderColor: this.props.error ? "red100" : "black10" })
    }
  }

  onBlur(e) {
    if (this.props.onBlur) {
      this.props.onBlur(e)
    }

    this.setState({
      borderColor: this.props.error ? "red100" : "black10",
    })
  }

  onFocus(e) {
    if (this.props.onFocus) {
      this.props.onFocus(e)
    }

    this.setState({ borderColor: "purple100" })
  }

  render() {
    return (
      <TextInput
        border={1}
        borderColor={this.state.borderColor}
        fontSize={3}
        p={3}
        pb={2}
        {...this.props}
        // These props should not be overridden so they are declared after `{...this.props}`
        ref={this.props.inputRef}
        onBlur={e => this.onBlur(e)}
        onFocus={e => this.onFocus(e)}
      />
    )
  }
}
