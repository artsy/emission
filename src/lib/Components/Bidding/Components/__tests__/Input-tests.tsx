import React from "react"
import * as renderer from "react-test-renderer"

import { BiddingThemeProvider } from "../BiddingThemeProvider"
import { Input } from "../Input"

it("renders properly", () => {
  const bg = renderer
    .create(
      <BiddingThemeProvider>
        <Input placeholder="Placeholder" />
      </BiddingThemeProvider>
    )
    .toJSON()

  expect(bg).toMatchSnapshot()
})
