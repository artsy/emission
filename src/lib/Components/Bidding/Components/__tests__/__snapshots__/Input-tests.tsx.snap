// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<TextInput
  allowFontScaling={true}
  border={1}
  borderColor="black10"
  fontSize={3}
  onBlur={[Function]}
  onFocus={[Function]}
  p={3}
  pb={2}
  placeholder="Placeholder"
  style={
    Array [
      Object {
        "borderColor": "#E5E5E5",
        "borderStyle": "solid",
        "borderWidth": 1,
        "fontFamily": "AGaramondPro-Regular",
        "fontSize": 16,
        "height": 40,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
      undefined,
    ]
  }
/>
`;
